File: graphs/130-Surrounded-Regions/neet.py
class Solution:
    def solve(self, board) -> None:
        rows, cols = len(board), len(board[0])
          
        def capture(r, c):
            if r < 0 or c < 0 or r == rows or c == cols or board[r][c] != "O":
                return
            board[r][c] = "U"
            capture(r + 1, c)  
            capture(r - 1, c)  
            capture(r, c + 1)  
            capture(r, c - 1)     

        # 1 capture unsurrounded (o -> u)
        for r in range(rows):
            for c in range(cols):
                if board[r][c] == "O" and (r in [0, rows - 1] or c in [0, cols - 1]):
                    capture(r, c)

        # 2 mark surrounded as x and convert u back to o
        for r in range(rows):
            for c in range(cols):
                if board[r][c] == "O":
                    board[r][c] = "X"
                if board[r][c] == "U":
                    board[r][c] = "O"               
                    

        

File: graphs/130-Surrounded-Regions/neet.js
// total sam

var solve = function (board) {
  const rows = board.length;
  const cols = board[0].length;

  const capture = (r, c) => {
    if (r < 0 || c < 0 || r === rows || c === cols || board[r][c] !== "O") {
      return;
    }

    board[r][c] = "U";
    capture(r + 1, c);
    capture(r - 1, c);
    capture(r, c + 1);
    capture(r, c - 1);
  };

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (
        board[r][c] === "O" &&
        ([0, rows - 1].includes(r) || [0, cols - 1].includes(c))
      ) {
        capture(r, c);
      }
    }
  }

  for (let r = 0; r < rows; r++) {
    for (let c = 0; c < cols; c++) {
      if (board[r][c] === "O") {
        board[r][c] = "X";
      }
      if (board[r][c] === "U") {
        board[r][c] = "O";
      }
    }
  }
};
